bin/i18n/curriculum_sync_utils.rb

```diff
@@ -0,0 +1,204 @@
+  #             "activity_sections": {
+  #               "ef4f13d4-c5b0-4778-8ca1-359132023d82": {
+  #                 "progression_name": "..."
+  #               }
+  #             }
+  #           },
+  #           ...
+  #         },
+  #         "resources": {
+  #           "url": {
+  #             "name": "..."
+  #           },
+  #         },
+  #       },
+  #       ...
+  #     },
+  #     "resources": {...}
+  #   },
+  #   "courseb-2021": {
+  #     "lessons": {...}
+  #     "resources": {...}
+  #   },
+  #   ...
+  # }
+  #
+  # return:
+  #
+  # {
+  #   "scripts": {
+  #     "coursea-2021": {...},
+  #     "courseb-2021": {...},
+  #     ...
+  #   },
+  #   "lessons": {
+  #     "http://studio.code.org/s/coursea-2021/lessons/1": {
+  #       "overview": "...",
+  #       "preparation": "...",
+  #       "purpose": "...",
+  #       "student_overview": "...",
+  #     },
+  #     "http://studio.code.org/s/courseb-2021/lessons/1": {...},
+  #     ...
+  #   },
+  #   "lesson_activities": {
+  #     "09eb0525-dfae-44f2-831f-5c1f2ffee133": {
+  #       "name": "...",
+  #     },
+  #     ...
+  #   },
+  #   "activity_sections": {
+  #     "ef4f13d4-c5b0-4778-8ca1-359132023d82": {
+  #       "progression_name": "..."
+  #     },
+  #     ...
+  #   },
+  #   "resources": {
+  #     "url": {
+  #       "name": "..."
+  #     },
+  #     ...
+  #   }
+  # }
+  def self.flatten(object_hash, serializer, name)
+    results = {}
+    object_hash.each do |key, object|
+      object = object.symbolize_keys
+
+      # store attributes directly on the results for this object
+      attributes = object.slice(*serializer._attributes)
+      if attributes.present?
+        results[name] ||= {}
+        results[name][key] = attributes
+      end
+
+      # recursively process "reflections" (ie, related model data) into their
+      # own objects
+      serializer._reflections.each do |reflection_key, reflection|
+        next unless object.key?(reflection_key) && object[reflection_key].present?
+        results.deep_merge!(flatten(object[reflection_key], reflection.options[:serializer], reflection.name))
+      end
+    end
+    return results
+  end
+
+  # The ScriptCrowdinSerializer used in the sync in organizes all content by
+  # script; each script is a single file, containing all the individual
+  # lessons, activities, etc, for the script. This works great for organizing
+  # content in a way that can easily be worked on by translators, but for
+  # rendering translations in model and view logic we're going to need to do
+  # something different.
+  #
+  # Specifically, we want to flatten the complex files generated by the
+  # sync in. Ultimately, we want one file per content type; one for all
+  # lessons, one for all activities, one for all resources, etc.
+  def self.sync_out_reorganize
+    Languages.get_locale.each do |prop|
+      locale = prop[:locale_s]
+      next if locale == 'en-US'
+      next unless locale == 'fr-FR'
```

Comment:
Is this left here on accident? I'm guessing you were using it for debugging purposes.
```suggestion
```

Reply:
right you are!